File: Optimization/Optimization (Single Case)/getDimLessNumbers.py
###############################################################################
# function:     getDimLessNumbers()                                           #
# Desctiption:  Equations to compute velocity, Reynolds Number, Schmidt       #
#               Number in both concentrate and dilute channels                #
#               Sherwood Numbers at (CEM/Concentrate), (AEM/Concentrate),     #
#               (CEM/Dilute), and (AEM/Dilute) interfaces                     #
#                                                                             #
# Input:        - m : Pyomo concrete model                                    #
#                                                                             #
# Output:       - m with Dimension-less numbers added as components           #
#                                                                             #
###############################################################################

def getDimLessNumbers(m): 
        
    # Velocity of the streams in concentrate and dilute channels
    m.uConc  = 0.5*(m.flowOutConcentrateND*m.scaleFacFlow + m.flowInConcentrate)/(m.memWidth*m.thicknessConcentrate)   
    m.uDil   = 0.5*(m.flowOutDiluteND*m.scaleFacFlow + m.flowInDilute)/(m.memWidth*m.thicknessDilute)  
    
    # Reynolds Numbers of the streams in concentrate and dilute channels
    m.ReConc = m.densityManure*m.uConc*(2*m.thicknessConcentrate)/m.viscosityManure       
    m.ReDil  = m.densityManure*m.uDil*(2*m.thicknessDilute)/m.viscosityManure   
    
    # Schmidt Numbers of the streams in concentrate and dilute channels
    m.ScConc = m.viscosityManure/(m.densityManure*m.saltDiffConc)
    m.ScDil  = m.viscosityManure/(m.densityManure*m.saltDiffDil )
    
    # Sherwood number of 4 Membrane/Channel Interfaces
    m.ShConcCEM = 0.29*(m.ReConc**0.5)*(m.ScConc**0.33)
    m.ShDilCEM  = 0.29*(m.ReDil **0.5)*(m.ScDil **0.33)
    m.ShConcAEM = 0.29*(m.ReConc**0.5)*(m.ScConc**0.33)
    m.ShDilAEM  = 0.29*(m.ReDil **0.5)*(m.ScDil **0.33)
    

File: Simulation/Simulation (Single Case)/getDimLessNumbers.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% function:    getDimLessNumbers(...)                                     %
%                                                                         %  
% Description: Compute DimensionLess numbers in the Electrodialysis cell  %
%                                                                         %
% Input:       flowConc - State Variable: Flowrate in concentrate channel % 
%              flowDil  - State Variable: Flowrate in concentrate channel %
%                                                                         %
% Output:      ReConc    - Reynolds Number in concentrate channel         %
%              ReDil     - Reynolds Number in dilute      channel         %
%              ScConc    - Schmidt Number  in concentrate channel         %
%              ScDil     - Schmidt Number  in dilute      channel         %
%              ShConcCEM - Sherwood Number in concentrate, CEM interface  %
%              ShConcAEM - Sherwood Number in concentrate, AEM interface  %
%              ShDilCEM  - Sherwood Number in dilute     , CEM interface  %
%              ShDilAEM  - Sherwood Number in dilute     , AEM interface  %
%                                                                         %
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
function [uConc, uDil, ReConc, ReDil, ScConc, ScDil, ShConcCEM, ShConcAEM, ShDilCEM, ShDilAEM] = getDimLessNumbers(flowConc, flowDil)

    % Get Params structure from the getParams function
    params        = getParams();

    % Unpack parameters
    densWater     = params.densWater     ;     % Density of Water (kg/m3)
    visWater      = params.visWater      ;     % Viscosity of Water (Pa-s)
    memWidth      = params.memWidth      ;     % Membrane Width  (m)
    diffConc      = params.diffConc      ;     % Diffusivity coefficient of ions in concentrate channel (m2/s)
    diffDil       = params.diffDil       ;     % Diffusivity coefficient of ions in dilute      channel (m2/s) 
    thicknessConc = params.thicknessConc ;     % Thickness of concentrate channel (m)
    thicknessDil  = params.thicknessDil  ;     % Thickness of dilute      channel (m)
    
    % Flow velocity in the concentrate and dilute channel
    uConc = flowConc/memWidth/thicknessConc/3600;
    uDil  = flowDil /memWidth/thicknessDil /3600;

    % Reynolds numbers in the concentrate and dilute channel
    ReConc = 2*densWater* flowConc/memWidth/visWater/3600;   % 3600 is used because flow is in m3/hr
    ReDil  = 2*densWater* flowDil /memWidth/visWater/3600;   % 3600 is used because flow is in m3/hr
    
    % Schmidt numbers in the concentrate and dilute channel
    ScConc = visWater/(densWater*diffConc);
    ScDil  = visWater/(densWater*diffDil );
    
    % Sherwood numbers in the interfaces of both membranes and fluids
    ShConcCEM = 0.29 * (ScConc^0.33 * ReConc^0.5);
    ShConcAEM = 0.29 * (ScConc^0.33 * ReConc^0.5);
    ShDilCEM  = 0.29 * (ScDil ^0.33 * ReDil ^0.5);
    ShDilAEM  = 0.29 * (ScDil ^0.33 * ReDil ^0.5);
    
    if ShConcAEM < 1
        ShConcAEM = 1;
        ShConcCEM = 1;
    end

end
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 